Demo/Tests/SnapshotTest/HostWindow.swift

```diff
@@ -0,0 +1,236 @@
+//  Copyright © 2017 One by AOL : Publishers. All rights reserved.
+
+import UIKit
+
+public class HostWindow: UIWindow {
+    
+    public var presentation: Presentation?
+    
+    init(presentation: Presentation) {
+        super.init(frame: CGRect.getCGRect(from: presentation.size))
+        self.presentation = presentation
+        presentation.mask?.frame = CGRect.getCGRect(from: presentation.size)
+        self.mask = presentation.mask
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public struct Presentation {
+        
+        let name: String
+        let size: CGSize
+        let traitCollection: UITraitCollection
+        let mask: UIView?
+        
+        public enum iPhone{
+            public enum X {
+                public static var portrait: Presentation {
+                    return Presentation(
+                        name: "iPhone_X_Portrait",
+                        size: CGSize.iPhone.X,
+                        traitCollection: UITraitCollection.iPhone.iPhoneRegular.portrait,
+                        mask: iPhoneXMask.portrait)
```

Comment:
Could this information be inherited from `enum X`?

Reply:
You mean instead of 
                        name: "iPhone_X_Portrait",
                        size:  CGSize(width: 375, height: 812),
                        traitCollection: UITraitCollection.iPhone.iPhoneRegular.portrait,
                        mask: iPhoneXMask.portrait
make it this way?
                        name: "iPhone_X_Portrait",
                        size: CGSize.iPhone.X,
                        traitCollection: UITraitCollection(
                                traitsFrom: [UITraitCollection(userInterfaceIdiom: .phone),
                                                    UITraitCollection(verticalSizeClass: .regular),
                                                    UITraitCollection(horizontalSizeClass: .compact)]),
                        mask: UIImageView(image: UIImage(named: "maskPortrait"))
The code of HostWindow class will become massive. Don't you think so?

Reply:
No - I meant different thing.

Reply:
Work in progress.

Reply:
Work in progress.